### PR TITLE
DBZ-7025 Debezium Operator blogpost

### PR DESCRIPTION
**Pull request by jcechace**
https://issues.redhat.com/browse/DBZ-7025

### CODE REVIEW

**Comment by jcechace** [comment]
Still needs some work (e.g. fix the publication date, links to files and so on) and proofreading

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+apiVersion: v1 <1>
+kind: Secret
+metadata:
+  name: postgresql-credentials
+type: opaque
+data:
+  POSTGRES_DB: ZGViZXppdW0=
+  POSTGRES_USER: ZGViZXppdW0=
+  POSTGRES_PASSWORD: ZGViZXppdW0=
+---
```
Comment (jcechace) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: @Naros Would you happen to know how to escape this? The syntax highlight is behaving strangely here.

Comment (mfvitale) [reply _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Have you tried with `\---` or `+---+`?
More details [here](https://docs.asciidoctor.org/asciidoc/latest/subs/prevent/) and [here](https://docs.asciidoctor.org/asciidoc/latest/pass/pass-macro/)

Comment (jcechace) [reply _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Yes, doesn't seem to work inside code block.

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+author: jcechace
+---
+
+With Debezium 2.3 we introduced the preview of the brand new Debezium Operator with the aim to provide seamless deployment of Debezium Server to Kubernetes (k8s) clusters. Debezium 2.4.0.Final release brings the next step towards the full support of this component. With this release we are happy to announce that Debezium Operator is now available in the OperatorHub https://operatorhub.io/operator/debezium-operator[catalog] for Kubernetes as well as in the community operator catalog embedded in OpenShift and OKD distributions. For now the operator still remains in the inclubation phase, however the full support of this component is approaching fast. 
+
++++<!-- more -->+++
+
+== The Goal 
+
+In this article we will demonstrate how to stream changes from PostgreSQL database into Apache Kafaka using Debezium Server deployed in Kubernetes cluster. We will also demonstrate some of the capabilities of our new k8s integration. For conveninece all code snippets and kubernetes manifests used in the tutorial are also available in our https://github.com/debezium/debezium-examples/tree/main/operator/tutorial-postgresql-kafka[GitHub repository for examples].
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. `conveninece` -> `convenience`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+
+== Preparing the Environment
+
+Before we get to deploying the operator and consequently Debezium Server itself, we first need an environment to deploy into. In this section we will showcase how to provision a local Kubernetes cluster with running PostgreSQL database and Apache Kafka broker. Note that it is not required for the database and/or the Kafka broker to run inside Kubernetes, we just chose to do so for the purpose of this demonstration.
+
+=== Running a Local Kubernetes Cluster
+You can skip this part if you already have a running Kubernetes cluster available, however make sure you posses a cluster-admin permissions as these are required for operator instalaltion. If not, then read on.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. 

1. `posses` -> `possess`
2. `instalaltion` -> `installation`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+
+There are multiple tools available to run a local k8s cluster, such as https://minikube.sigs.k8s.io/docs/[Minikube], https://kind.sigs.k8s.io/[Kind], or https://docs.docker.com/desktop/kubernetes/[Docker Desktop]. In ths article we will be using Kind to create a local single node cluster.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. `ths` -> `this`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+[source, bash]
+----
+kubectl create namespace debezium
+----
+
+=== Deploying the infrastructure
+In this section we will prepare the required infrastructure -- PostgreSQL database as well as an instance of Kafka broker. 
+
+==== PostgreSQL Database 
+Lets start with deploying the PostgreSQL database by executing the command bellow.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. `bellow` -> `below`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+
+==== PostgreSQL Database 
+Lets start with deploying the PostgreSQL database by executing the command bellow. 
+
+[CAUTION]
+For simplicity we are using an ephemeral volume mounts which means that any modification done to our database instance will not persist when the pod is recreated. 
+
+[source, bash]
+----
+kubect create -f https://raw.githubusercontent.com/debezium/debezium-examples/main/operator/tutorial-postgresql-kafka/infra/001_postgresql.yml -n debezium
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: typo. `kubect` -> `kubectl`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+$ kubectl get deployments -n debezium
+
+NAME                        READY   UP-TO-DATE   AVAILABLE
+postgresql                  1/1     1            1
+----
+
+==== Kafka Broker
+To deploy a Kafka broker instance we will take an advantage of the https://strimzi.io/[Strimzi Operator]. 
+
+So first we will deploy the Strimzi operator itself by running the command bellow. Please not the `namespace` parameter in the URL -- it's importnat as it ensures that Kubernetes objects required by Strimzi are created in the correct namespace.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. 

1. `bellow` -> `below`
2. `Please not` -> `Please note` ?

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 8]
```diff
@@ -0,0 +1,433 @@
...
+
+
+=== Deploying Debezium Operator from Operator Catalog
+In this section we will use the https://olm.operatorframework.io/[Operator Lifecycle Manager] to create a subscription to an operator available in the https://operatorhub.io/[OperatorHub catalog]. As we mentioned previously, Debezium is one of the https://operatorhub.io/operator/debezium-operator[available operators].
+
+Besides other things, using OLM also allows you to configure the scope of namespaces watched by the operator from a single namespace to the entire cluster. However https://olm.operatorframework.io/docs/advanced-tasks/operator-scoping-with-operatorgroups/[this configuration] is out of the scope (pun intended!) for this tutorial. The process bellow will install the operator into the `operators` namespace -- which is by default intended for cluster-wide operators.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. `bellow` -> `below`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 9]
```diff
@@ -0,0 +1,433 @@
...
+
+
+=== Deploying Debezium Operator from Operator Catalog
+In this section we will use the https://olm.operatorframework.io/[Operator Lifecycle Manager] to create a subscription to an operator available in the https://operatorhub.io/[OperatorHub catalog]. As we mentioned previously, Debezium is one of the https://operatorhub.io/operator/debezium-operator[available operators].
+
+Besides other things, using OLM also allows you to configure the scope of namespaces watched by the operator from a single namespace to the entire cluster. However https://olm.operatorframework.io/docs/advanced-tasks/operator-scoping-with-operatorgroups/[this configuration] is out of the scope (pun intended!) for this tutorial. The process bellow will install the operator into the `operators` namespace -- which is by default intended for cluster-wide operators. 
+
+First we need to install OLM itself by running the following shell commands -- skip this if OLM is already instaled in your cluster.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. `instaled` -> `installed`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 10]
```diff
@@ -0,0 +1,433 @@
...
+----
+
+[TIP]
+In previous section we chose to deploy Strimzi operator by directly applying a set of k8s manifests to our cluster. However Strimzi is also one of the operators available in OperatorHub catalog and as such could be also installed via OLM. 
+
+=== Using Raw Kubernetes Manifests to Deploy Debezium Operator
+This options allows deployment of Debezium Operator into any Kubernetes cluster without the need for OLM. 
+
+[IMPORTANT]
+Debezium Operator depoyed this way will be limitted to managing Debezium Server instances **only in the same namespace as the operator**.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo.

1. `depoyed` -> `deployed`
2. `limitted` -> `limited`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 11]
```diff
@@ -0,0 +1,433 @@
...
+<1> Resource type monitored by Debezium Operator.
+<2> Name of the deployed Debezium Server instance.
+<3> Optional property specifying the container image .
+<4> Quarkus configuration used by Debezium Server.
+<5> Kafka sink configuration.
+<6> PostgreSQL source connector configuration.
+
+The `spec` part of the manifest will likely be familiar to anybody with previous Debezium Server experience as it is a more structured variant of Debezium Server property configuration. In our case the `image` property is particularly redundant as it uses the default image for installed operator version. 
+
+The quarkus part of the `spec` provides Debezium Server with access to previously deployed `postgresql-credentials` secret containg the credentials to our database. You can see the `POSTGRES_USER` and other variables referenced later on in the configuration.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. `containg` -> `containing`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 12]
```diff
@@ -0,0 +1,433 @@
...
+The `spec` part of the manifest will likely be familiar to anybody with previous Debezium Server experience as it is a more structured variant of Debezium Server property configuration. In our case the `image` property is particularly redundant as it uses the default image for installed operator version. 
+
+The quarkus part of the `spec` provides Debezium Server with access to previously deployed `postgresql-credentials` secret containg the credentials to our database. You can see the `POSTGRES_USER` and other variables referenced later on in the configuration. 
+ 
+
+
+A bit more detailed description of the `DebeziumServer` custom resource can be found at https://github.com/debezium/debezium-operator/tree/2.4#debeziumserverspec-reference[GitHub]. 
+
+=== Under the Hood
+Debezium Operator will take care of creating everything required to run Debezium Server insinde Kubernetes.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. `insinde` -> `inside`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 13]
```diff
@@ -0,0 +1,433 @@
...
+
+[source, bash]
+----
+$ kubectl get deployments -n debezium
+
+NAME                        READY   UP-TO-DATE   AVAILABLE 
+my-debezium                 1/1     1            1 
+----
+
+Witht Debezium Server running we can verify that it consumed all initial data from the database with the following command.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. `Witht` -> `With`

File: _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc [hunk 14]
```diff
@@ -0,0 +1,433 @@
...
+    --bootstrap-server localhost:9092 \
+    --from-beginning \
+    --property print.key=true \
+    --topic inventory.inventory.orders
+----
+
+
+== The Future and Our Request
+
+This is it for now. Before the operator gets full support we intend to provied a more detailed documentation as well as the ability to further configure the deployment with things such as custom pull secrets in order to suport customized Debezium Server images stored in secured registries.
```
Comment (ilyasahsan123) [inline _posts/2023-10-10-Debezium-Operator-Takes-off-to-the-Clouds.adoc]: Typo. 

1. `provied` -> `provide`
2. `suport` -> `support`

**Comment by jcechace** [comment]
@ilyasahsan123  once again, thanks!

**Comment by jcechace** [comment]
@Naros this is ready for a review now, please have a look when you get the time -- you are a native speaker after all :)

**Pull request merged**